### PR TITLE
Add deprecation warnings for aws-sdk < 2 and active record < 4.2

### PR DESCRIPTION
**Pull request by kevcha**
Related to #2049, this commit
* Adds a deprecation warning if the user is still on aws-sdk v1
* Adds a deprecation warning if the user is still on Rails 3.2.x, say 3.2.x support will be dropped in v5.0.0

### CODE REVIEW

File: lib/paperclip/deprecations.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require "active_support/deprecation"
+
+module Paperclip
+  class Deprecations
```
Comment (tute) [inline lib/paperclip/deprecations.rb]: Suggestion for this file (including a tweaked warning message):

```ruby
require "active_support/deprecation"

module Paperclip
  class Deprecations
    class << self
      def check
        warn_aws_sdk_v1 if aws_sdk_v1?
        warn_outdated_rails if old_active_record?
      end

      private

      def aws_sdk_v1?
        defined?(::Aws) && ::Aws::VERSION < "2"
      end

      def old_active_record?
        ::ActiveRecord::VERSION::STRING < "4.2"
      end

      def warn_aws_sdk_v1
        warn "[paperclip] [deprecation] AWS SDK v1 has been deprecated in paperclip 5. Please consider upgrading to AWS 2 before upgrading paperclip."
      end

      def warn_outdated_rails
        warn "[paperclip] [deprecation] Rails < 4.2 are unsupported as of Rails 5 release. Please upgrade to Rails 4.2 or 5 before upgrading paperclip."
      end

      def warn
        ActiveSupport::Deprecation.warn
      end
    end
  end
end
```

Comment (tute) [inline lib/paperclip/deprecations.rb]: My version is not taking into account the stubs, please ignore those changes. Thank you!

File: lib/paperclip/deprecations.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      def aws_sdk_v1?
+        defined?(::Aws) && aws_sdk_version < "2"
+      end
+
+      def warn_aws_sdk_v1
+        ActiveSupport::Deprecation.warn "[paperclip] [deprecation] AWS SDK v1 has been deprecated in paperclip 5. Please consider upgrading to AWS 2 before upgrading paperclip."
+      end
+
+      def warn_outdated_rails
+        ActiveSupport::Deprecation.warn "[paperclip] [deprecation] Rails 3.2 and 4.1 are unsupported as of Rails 5 release. Please upgrade to Rails 5 before upgrading paperclip."
```
Comment (tute) [inline lib/paperclip/deprecations.rb]: Should this read "please ugpgrade to 4.2" instead of 5?

**Comment by tute** [comment]
Looks great, thank you! Left one small question around a warning message, ready for merge after we address that.

File: lib/paperclip/deprecations.rb (hunk 2 above)
Comment (kevcha) [inline lib/paperclip/deprecations.rb]: :+1:

File: lib/paperclip/deprecations.rb (hunk 1 above)
Comment (kevcha) [inline lib/paperclip/deprecations.rb]: :+1:, That's better, I'll update the stubs

**Comment by kevcha** [comment]
I added a commit taking in account your comments.
I'll squash it when everything will be good (to avoid spamming related issue...)

**Comment by maclover7** [comment]
Awesome work @kevcha! Just a few Hound CI comments, and then you're ready for merge.

**Comment by maclover7** [comment]
@tute - does this PR have to be open against the v4.3 branch?

**Comment by tute** [comment]
> @tute - does this PR have to be open against the v4.3 branch, so that v4.4.0 can be released from that? Master is already tracking v5.0.0.

Indeed, thanks for catching that. Can you reopen the PR against that branch, @kevcha?

**Comment by maclover7** [comment]
Cool - @kevcha, can you please open up this PR against the `v4.3` branch please? The master branch already has 5.0 changes in it :)

**Pull request closed without merging**